Marlin/Marlin_main.cpp

```diff
@@ -1767,215 +1762,347 @@ static void clean_up_after_endstop_or_probe_move() {
       }
     #endif
 
-    if (axis_unhomed_error(true, false, false)) return;
-
-    float oldXpos = current_position[X_AXIS]; // save x position
-
     // Dock sled a bit closer to ensure proper capturing
     do_blocking_move_to_x(X_MAX_POS + SLED_DOCKING_OFFSET - ((stow) ? 1 : 0));
     digitalWrite(SLED_PIN, !stow); // switch solenoid
 
-    do_blocking_move_to_x(oldXpos); // return to position before docking
-
   }
 
 #endif // Z_PROBE_SLED
 
-#if HAS_BED_PROBE
-
-  static void deploy_z_probe() {
-
-    #if ENABLED(DEBUG_LEVELING_FEATURE)
-      if (DEBUGGING(LEVELING)) DEBUG_POS("deploy_z_probe", current_position);
+#if ENABLED(Z_PROBE_ALLEN_KEY)
+  void deploy_allen_key() {
+    #if defined(Z_PROBE_ALLEN_KEY_DEPLOY_1_X) || defined(Z_PROBE_ALLEN_KEY_DEPLOY_1_Y) || defined(Z_PROBE_ALLEN_KEY_DEPLOY_1_Z)
+      do_blocking_move_to(
+      #ifdef Z_PROBE_ALLEN_KEY_DEPLOY_1_X
+        Z_PROBE_ALLEN_KEY_DEPLOY_1_X,
+      #else
+        current_position[X_AXIS],
+      #endif
```

Comment:
This code gets pretty long and unwieldy pretty quickly. We can save 2 lines per-instance of these defines, and make the resulting code look nicer if we do this ahead of time, instead of doing these inline…
```cpp
#ifndef Z_PROBE_ALLEN_KEY_DEPLOY_1_X
  #define Z_PROBE_ALLEN_KEY_DEPLOY_1_X current_position[X_AXIS]
#endif
#ifndef Z_PROBE_ALLEN_KEY_DEPLOY_1_Y
  #define Z_PROBE_ALLEN_KEY_DEPLOY_1_Y current_position[Y_AXIS]
#endif
. . .
```